great_expectations/data_context/data_context.py

```diff
@@ -1165,7 +1133,9 @@ def list_validation_operator_names(self):
         event_name="data_context.add_datasource",
         args_payload_fn=add_datasource_usage_statistics,
     )
-    def add_datasource(self, name, initialize=True, **kwargs):
+    def add_datasource(
+        self, name, initialize=True, **kwargs
+    ) -> Optional[Dict[str, Union[LegacyDatasource, BaseDatasource]]]:
```

Comment:
Why is the return type here `Optional`?

Reply:
This is to capture that it can sometimes be None (otherwise PyCharm points out signature/variable mismatches).  The syntax `Optional[some_type]` works as a nice shorthand for `Union[some_type, None]`.

Reply:
Capturing my learning here: this is because if initialize is set to false, then we *want* to return None